basic/src/test/java/org/jboss/hal/testsuite/test/configuration/singleton/ElectionPolicyTestCase.java

```diff
@@ -0,0 +1,206 @@
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class ElectionPolicyTestCase {
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private SingletonSubsystemPage page;
+
+    private static final String
+            SINGLETON_POLICY_RESOURCE = "s-pol_" + RandomStringUtils.randomAlphanumeric(5),
+            SINGLETON_POLICY_EDIT_RESOURCE = "s-pol-edit_" + RandomStringUtils.randomAlphanumeric(5),
+            CACHE_CONTAINER_RESOURCE = "cache-container-singleton_" + RandomStringUtils.randomAlphanumeric(5),
+            CACHE_CONTAINER = "cache-container",
+            SINGLETON_POLICY = "singleton-policy",
+            ELECTION_POLICY = "election-policy",
+            NAME_PREFERENCES = "name-preferences",
+            SOCKET_BINDING_PREFERENCES = "socket-binding-preferences",
+            SIMPLE = "simple";
+
+    private static final Address
+            SINGLETON_SUBSYSTEM_ADDRESS = ConfigUtils.isDomain() ? Address.of("profile", "full-ha").and("subsystem", "singleton") : Address.subsystem("singleton"),
+            INFINISPAN_SUBSYSTEM_ADDRESS = ConfigUtils.isDomain() ? Address.of("profile", "full-ha").and("subsystem", "infinispan") : Address.subsystem("infinispan"),
+            SINGLETON_POLICY_EDIT_ADDRESS = SINGLETON_SUBSYSTEM_ADDRESS.and("singleton-policy", SINGLETON_POLICY_EDIT_RESOURCE),
+            SINGLETON_POLICY_ADDRESS = SINGLETON_SUBSYSTEM_ADDRESS.and("singleton-policy", SINGLETON_POLICY_RESOURCE),
+            CACHE_CONTAINER_RESOURCE_ADDRESS = INFINISPAN_SUBSYSTEM_ADDRESS.and("cache-container", CACHE_CONTAINER_RESOURCE);
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final Administration administration = new Administration(client);
+    private static final Operations operations = new Operations(client);
+
+    private static SocketBindingResources socketBindingResources;
+
+    @Before
+    public void before() {
+        page.navigate();
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws IOException, CommandFailedException {
+        socketBindingResources = new SocketBindingResources(client, 3, "election-policy_");
+        SingletonSubsystemOperations.withOperations(operations).prepareCacheContainer(CACHE_CONTAINER_RESOURCE_ADDRESS);
+        operations.add(SINGLETON_POLICY_ADDRESS, Values.of(CACHE_CONTAINER, CACHE_CONTAINER_RESOURCE)).assertSuccess();
+        operations.add(SINGLETON_POLICY_EDIT_ADDRESS, Values.of(CACHE_CONTAINER, CACHE_CONTAINER_RESOURCE)).assertSuccess();
+        operations.add(SINGLETON_POLICY_EDIT_ADDRESS.and(ELECTION_POLICY, "simple")).assertSuccess();
```

Comment:
I suggest using batch operation, which takes care of cleaning after itself if not all of the substeps succeed.